lib/init.php

```diff
@@ -148,3 +148,49 @@ function register_site_logo_to_rest_index( $response ) {
 }
 
 add_filter( 'rest_index', 'register_site_logo_to_rest_index' );
+
+/**
+ * Optimizes the preload paths registered in Core (`edit-form-blocks.php`).
+ *
+ * @param array $preload_paths Preload paths to be filtered.
+ * @return array
+ */
+function optimize_preload_paths( $preload_paths ) {
```

Comment:
This function filters patches the `preload_paths` list [created in Core](https://github.com/jsnajdr/wordpress-develop/blob/trunk/src/wp-admin/edit-form-blocks.php#L59-L70) when loading the block editor page.

When a Core patch with these changes is merged, the filter becomes a noop. The Core list and the Gutenberg plugin will always be out of sync, so this filter will patch the list according to latest Gutenberg's requirements.

Reply:
You should move this change to `lib/wordpress-6.0/edit-form-blocks.php`, that way it's easier for us to know when to remove that code from the plugin (minimum supported version) and how to backport it to Core (same file name).

We should probably do the same for the file above but I guess since it was not done before for the remaining parts of that file, it's fine to keep your changes there for now.

Reply:
Good idea, moved the `edit-form-blocks` bits to `lib/compat`.

About the `edit-site-page.php` modifications: there's not a direct counterpart to that file in Core. The Gutenberg plugins ships a completely standalone page for Full Site Editing, one that's not interacting or integrated with the one in Core at all. Its URL is `/wp-admin/themes.php?page=gutenberg-edit-site`, while the Core equivalent is `/wp-admin/site-editor.php`. The Gutenberg plugin intercepts all the links and replaces them.

Reply:
We'll drop using custom page once Gutenberg plugin requires WP 5.9 or later.